src/neo-vm/Types/Buffer.cs

```diff
@@ -18,40 +19,53 @@ namespace Neo.VM.Types
     /// <summary>
     /// Represents a memory block that can be used for reading and writing in the VM.
     /// </summary>
-    [DebuggerDisplay("Type={GetType().Name}, Value={System.BitConverter.ToString(InnerBuffer).Replace(\"-\", string.Empty)}")]
+    [DebuggerDisplay("Type={GetType().Name}, Value={System.BitConverter.ToString(GetSpan().ToArray()).Replace(\"-\", string.Empty)}")]
     public class Buffer : StackItem
     {
         /// <summary>
         /// The internal byte array used to store the actual data.
         /// </summary>
-        public readonly byte[] InnerBuffer;
+        public readonly Memory<byte> InnerBuffer;
 
         /// <summary>
         /// The size of the buffer.
         /// </summary>
         public int Size => InnerBuffer.Length;
         public override StackItemType Type => StackItemType.Buffer;
 
+        private readonly byte[] _buffer;
+        private bool _keep_alive = false;
+
         /// <summary>
         /// Create a buffer of the specified size.
         /// </summary>
         /// <param name="size">The size of this buffer.</param>
         /// <param name="zeroInitialize">Indicates whether the created buffer is zero-initialized.</param>
         public Buffer(int size, bool zeroInitialize = true)
         {
-            InnerBuffer = zeroInitialize
-                ? new byte[size]
-                : GC.AllocateUninitializedArray<byte>(size);
+            _buffer = ArrayPool<byte>.Shared.Rent(size);
```

Comment:
I think that it won't use `AllocateUninitializedArray` so it will be zero initialized already